akka-docs-dev/rst/scala/http/routing-dsl/index.rst

```diff
@@ -74,3 +74,25 @@ Failures and exceptions inside the Routing DSL
 Exception handling within the Routing DSL is done by providing :class:`ExceptionHandler` s which are documented in-depth
 in the :ref:`exception-handling-scala` section of the documtnation. You can use them to transform exceptions into
 :class:`HttpResponse` s with apropriate error codes and human-readable failure descriptions.
+
+File uploads
+^^^^^^^^^^^^
+
+Handling a simple file upload from for example a browser form with a `file` input can be done
+by accepting a `Multipart.FormData` entity, note that the body parts are `Source` rather than
+all available right away, and so is the individual body part payload so you will need to consume
+those streams both for the file and for the form fields.
+
+Here is a simple example which just dumps the uploaded file into a temporary file on disk, collects
```

Comment:
"which streams the uploaded bytes into a file"? Need to stress that it's still streaming (people were confused when I showed a file in an example)